checks/check-slither.ts

```diff
@@ -0,0 +1,78 @@
+import { writeFileSync, unlinkSync } from 'fs'
+import util from 'util'
+import { exec as execCallback } from 'child_process'
+import { getContractName } from '../utils/clients/tenderly'
+import { ProposalCheck } from '../types'
+
+// convert exec method from a callback to a promise
+const exec = util.promisify(execCallback)
+
+// Data returned from command execution
+type ExecOutput = {
+  stdout: string
+  stderr: string
+}
+
+/**
+ * Runs slither against the verified contracts and reports the outputs.
+ * Assumes slither is already installed and can be run with python
+ */
+export const checkSlither: ProposalCheck = {
+  name: 'Runs slither against the verified contracts',
+  async checkProposal(proposal, sim, deps) {
+    let info = ''
+    let warnings: string[] = []
+    // For each verified contract with code, we write the files, run slither, then delete the
+    // files. We do this instead of running Slither on all contracts at once since contracts at
+    // different addresses can have the same names, which would result in an error when compiling
+    for (const contract of sim.contracts) {
+      // Get solc version used
+      const solcVersionMatch = contract.compiler_version.match(/\d*\.\d*\.\d*/)
+      if (!solcVersionMatch) {
+        const msg = `Slither not run for \`${contract.contract_name}\` at \`${contract.address}\`: could not parse solc version`
+        warnings.push(msg)
+        continue
+      }
+      const solcVersion = solcVersionMatch[0]
+
+      // Save the contract locally
+      for (const file of contract.data.contract_info) writeFileSync(file.name, file.source)
```

Comment:
we're writing them all to `.`? maybe better to put in a temp dir

Reply:
done in 7e40b35